Implement counter-style additive-symbols translation

vitorroriz opened the pull request:
#### 5c08dda8b880c88f13e7c9ff3139c3cfcc59ccd7
<pre>
Implement counter-style additive-symbols translation
<a href="https://bugs.webkit.org/show_bug.cgi?id=252926">https://bugs.webkit.org/show_bug.cgi?id=252926</a>
rdar://103837257
Reviewed by Darin Adler and Tim Nguyen.

We are missing translation of the counter-styles
additive-symbols for the CSSCounterStylesDescriptor.

We are also introducing some fixes and style
adequation to the other translation functions.

* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
(WebCore::translateRangeFromStyleProperties):
(WebCore::symbolToString):
(WebCore::translateAdditiveSymbolsFromStyleProperties):
(WebCore::translatePadFromStyleProperties):
(WebCore::translateNegativeSymbolsFromStyleProperties):
(WebCore::translateSymbolsFromStyleProperties):
(WebCore::translateFallbackNameFromStyleProperties):
(WebCore::translatePrefixFromStyleProperties):
(WebCore::translateSuffixFromStyleProperties):
(WebCore::extractDataFromSystemDescriptor):
(WebCore::CSSCounterStyleDescriptors::create):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeCounterStyleAdditiveSymbols):

Canonical link: <a href="https://commits.webkit.org/260912@main">https://commits.webkit.org/260912@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0029748bc75084f509a482834ef777031bc62b3d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109932 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19038 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/42622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1380 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/118990 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113882 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20500 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10226 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102194 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115681 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15261 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/42622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/43478 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/42622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/85290 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11742 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/42622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/12367 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/17736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/42622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7566 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14159 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

vitorroriz commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,6 +67,28 @@ static String symbolToString(const CSSValue* value)
+            ASSERT(weightValue.isPrimitiveValue() && symbolValue.isPrimitiveValue());
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
+            ASSERT(weight.isInteger());
+            result.constructAndAppend(symbolToString(&symbolValue), static_cast<unsigned>(downcast<CSSPrimitiveValue>(weightValue).intValue()));
```
ops, forgot to pass weight, instead of downcasting again

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,6 +67,28 @@ static String symbolToString(const CSSValue* value)
+    auto& list = downcast<CSSValueList>(*additiveSymbols);
+    CSSCounterStyleDescriptors::AdditiveSymbols result;
+    for (auto& additiveValue : list) {
+        if (auto* additiveTuple = dynamicDowncast<CSSValueList>(&additiveValue)) {
```
We should really be using CSSValuePair here: https://searchfox.org/wubkat/rev/98761d6864f7dac2f22655a51bb8a00a2172edc3/Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp#8227

like consumeCounterStyleRange does

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,6 +67,28 @@ static String symbolToString(const CSSValue* value)
+            auto& weightValue = (*additiveTuple)[0];
+            auto& symbolValue = (*additiveTuple)[1];
+            ASSERT(weightValue.isPrimitiveValue() && symbolValue.isPrimitiveValue());
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
```
@darinadler Should this be using one of the fromCSSValue* helpers?

darinadler replied:
Those are for where we don’t want to have to name the type explicitly, like in generated code, and want to deduce the type based on the destination. Or for when we want to translate a `valueID` into an enumeration. I would suggest `value<unsigned>` rather than `intValue`, though.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,6 +67,27 @@ static String symbolToString(const CSSValue* value)
+        if (auto* additiveTuple = dynamicDowncast<CSSValuePair>(&additiveValue)) {
+            auto& weightValue = additiveTuple->first();
+            auto& symbolValue = additiveTuple->second();
+            ASSERT(weightValue.isPrimitiveValue() && symbolValue.isPrimitiveValue());
```
It’s unnecessary and wasteful to assert when we are using `downcast`. The point of `downcast` is that it incorporates the assertion.

vitorroriz replied:
will fix, thanks!

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,6 +67,27 @@ static String symbolToString(const CSSValue* value)
+            auto& symbolValue = additiveTuple->second();
+            ASSERT(weightValue.isPrimitiveValue() && symbolValue.isPrimitiveValue());
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
+            ASSERT(weight.isInteger());
```
Is this an important assertion?

vitorroriz replied:
If it is not an integer the parser should have invalidated the rule before the translation but I thought it would be good to assert here to make sure.

nt1m replied:
I think we can remove this assert in that case.

darinadler replied:
I think it’s good to have assertions that the code actually relies on, and that often happens naturally due to assertions built into things like `downcast`, but here we just rely on being able to **convert** to an integer, so it’s probably overkill to additionally assert.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -36,7 +36,7 @@ namespace WebCore {
 static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(const StyleProperties& properties)
 {
-    auto ranges = properties.getPropertyCSSValue(CSSPropertySystem).get();
+    auto ranges = properties.getPropertyCSSValue(CSSPropertyRange).get();
```
Do we have test coverage for this mistake? I don’t see a change to a test result.

nt1m replied:
RenderListMarker integration + https://github.com/WebKit/WebKit/pull/10644  are needed before we can see any kind of test coverage changes. Though I'm very confident WPT has some visual coverage for this.

darinadler replied:
Same `get()` problem here. Pre-existing but wrong.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,6 +67,23 @@ static String symbolToString(const CSSValue* value)
+static CSSCounterStyleDescriptors::AdditiveSymbols translateAdditiveSymbolsFromStyleProperties(const StyleProperties& properties)
+{
+    auto additiveSymbols = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols).get();
+    auto& list = downcast<CSSValueList>(*additiveSymbols);
```
Do we need a null check here? Didn't check, but this _might_ be why the patch is causing crashes

darinadler replied:
That’s a good point. And I see another problem here. We should not store the result of `get()` in a local variable if the thing we are calling it on is not in one. We need to store the result of `getPropertyCSSValue` in a local.

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
```diff
@@ -8287,7 +8287,7 @@ RefPtr<CSSValue> consumeCounterStyleAdditiveSymbols(CSSParserTokenRange& range,
         lastWeight = weight;
 
-        return CSSValueList::createSpaceSeparated(integer.releaseNonNull(), symbol.releaseNonNull());
+        return CSSValuePair::create(integer.releaseNonNull(), symbol.releaseNonNull());
```
Switching from a list to a pair here requires changing the code that interprets this value to access it as a pair rather than a list. I don’t see that change here. I think that’s another possible reason for the crashes.

vitorroriz replied:
I'll check that up but regarding the crash itself, the code that actually uses the parsed counter-style shouldn't be reachable by now, since the fragment wiring it to RenderListMarker is in a separated PR.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +67,78 @@ static String symbolToString(const CSSValue* value)
+        return { };
+    CSSValueList* list;
+    if (!(list = downcast<CSSValueList>(additiveSymbols.get())))
+        return { };
```
```suggestion
    auto additiveSymbols = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols);
    auto* list = dynamicDowncast<CSSValueList>(additiveSymbols.get());
    if (!list)
        return { };
```

(Same above with ranges)

vitorroriz replied:
this doesn't seem right, we should check for `!additiveSymbols`, but I can do the `(!list)` check on a separated line if that's what you mean

darinadler replied:
Missing properties happen. A property with a value of the wrong type can’t happen. The two checks aren’t equivalent. One is needed, the other is dead code and we’d be better off with an assert.

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -961,11 +961,11 @@ CSSCounterStyleAtRulesEnabled:
+      default: true
     WebCore:
-      default: false
+      default: true
```
If this being set to false is what was causing the crash, we should fix the preference to properly work when disabled as well.

vitorroriz replied:
I've turned on locally for testing on minibrowser and added it by accident, but it is already removed.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -36,13 +36,14 @@ namespace WebCore {
-    auto& list = downcast<CSSValueList>(*ranges);
+    auto* list = dynamicDowncast<CSSValueList>(ranges.get());
+    if (!list)
+        return { }; // auto range will return an empty Ranges
```
Can we remove this comment? it doesn't match the webkit style and doesn't add much value.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+    if (!symbolsValues)
+        return { };
+    CSSValueList* list;
+    if (!(list = downcast<CSSValueList>(symbolsValues.get())))
```
Can we rename this variable to `symbols` ?
```suggestion
    CSSValueList* symbols = dynamicDowncast<CSSValueList>(value.get());
    if (!symbols)
```

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
-    if (!symbolsValues || !is<CSSValueList>(symbolsValues))
+    auto symbolsValues = properties.getPropertyCSSValue(CSSPropertySymbols);
+    if (!symbolsValues)
+        return { };
```
Can we just name the variable `value` while we're here?

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
     Vector<CSSCounterStyleDescriptors::Symbol> result;
-    auto& list = downcast<CSSValueList>(*symbolsValues);
-    for (auto& symbolValue : list) {
+    for (auto& symbolValue : *list) {
```
Ditto here with name:
```suggestion
    for (auto& symbol : *symbols) {
```

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+    for (auto& symbolValue : *list) {
         auto symbolString = symbolToString(&symbolValue);
         if (!symbolString.isNull())
             result.append((symbolString));
```
Can we rename `symbolString` to `string` ?

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -121,39 +149,39 @@ static Vector<CSSCounterStyleDescriptors::Symbol> translateSymbolsFromStylePrope
+    auto systemValue = systemCSSValue.get();
     // If no value is provided after `fixed`, the first synbol value is implicitly 1 (https://www.w3.org/TR/css-counter-styles-3/#first-symbol-value).
-    if (!systemValue || !systemValue->isPrimitiveValue())
+    if (!systemValue)
```
In this case, this will just work fine:
```
auto systemValue = properties.getPropertyCSSValue(CSSPropertySystem);
// ...
if (!systemValue || !systemValue->isPrimitiveValue())
```

As long as we define the the local variable once as a RefPtr, it should give us lifetime guarantees.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+        return { };
+    CSSValueList* list;
+    if (!(list = downcast<CSSValueList>(pad.get())))
+        return { };
```
We should probably fix this in another patch, but this should use CSSValuePair
```suggestion
    CSSValueList* list = dynamicDowncast<CSSValueList>(pad.get());
    if (!list)
        return { };
```

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+{
+    auto pad = properties.getPropertyCSSValue(CSSPropertyPad);
+    if (!pad)
+        return { };
```
Can we name this variable `value`?

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
     if (!length)
         return CSSCounterStyleDescriptors::Pad();
     auto lengthValue = length->intValue();
     ASSERT(lengthValue >= 0);
```
I think we can safely assume the length is a primitive value.
```
    auto length = downcast<CSSPrimitiveValue>(list->item(0))->intValue();
    ASSERT(length >= 0);
```

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+            auto& weightValue = additiveTuple->first();
+            auto& symbolValue = additiveTuple->second();
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
+            result.constructAndAppend(symbolToString(&symbolValue), weight.value<unsigned>());
```
```suggestion
        if (auto* pair = dynamicDowncast<CSSValuePair>(&value)) {
            auto weight = downcast<CSSPrimitiveValue>(pair->first());
            auto symbol = symbolToString(&pair->second());
            result.constructAndAppend(symbol, weight.value<unsigned>());
```

darinadler replied:
Should be `auto& weight`.

darinadler replied:
I would just write `auto weight = downcast<CSSPrimitiveValue>(pair->first()).value<unsigned>();`

vitorroriz replied:
yes, I've noticed the copy constructor is deleted. I'll follow the suggestion, thanks.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+        return { };
+    auto* list = downcast<CSSValueList>(additiveSymbols.get());
+    if (!list)
+        return { };
```
There's not much point in a null check here, unless we use `dynamicDowncast`. `downcast` assumes the value is already a CSSValueList.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,93 +68,112 @@ static String symbolToString(const CSSValue* value)
+    if (!value)
+        return { };
+
+    CSSValueList* list = downcast<CSSValueList>(value.get());
```
Better idiom: `auto& list = downcast<CSSValueList>(*value);`

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,93 +68,112 @@ static String symbolToString(const CSSValue* value)
+
+    CSSValueList* list = downcast<CSSValueList>(value.get());
+    ASSERT(list->size() == 2);
+    auto length = downcast<CSSPrimitiveValue>(list->item(0))->intValue();
```
And then `auto length = downcast<CSSPrimitiveValue>(list[0]).intValue();`

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,93 +68,112 @@ static String symbolToString(const CSSValue* value)
+
+    auto* list = downcast<CSSValueList>(additiveSymbols.get());
+    CSSCounterStyleDescriptors::AdditiveSymbols result;
+    for (auto& value : *list) {
```
```suggestion
    auto& list = downcast<CSSValueList>(*additiveSymbols);
    CSSCounterStyleDescriptors::AdditiveSymbols result;
    for (auto& value : list) {
```

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,93 +68,112 @@ static String symbolToString(const CSSValue* value)
+        return result;
+        }
     }
+    result.m_prefix = symbolToString(negative.get());
```
The consumeCounterStyleNegative function seems to suggest we always get a CSSValueList. 

```
auto& list = downcast<CSSValueList>(*negative);
ASSERT(list.size() == 1 || list.size() == 2);
result.m_prefix = symbolToString(list.item(0));
if (list.size() == 2)
  result.m_suffix = symbolToString(list.item(1));
return result;
```

vitorroriz replied:
it can also return a single symbol if range is over after consuming first token (or if consuming token fails)

nt1m replied:
Ah right, I missed that, sorry.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -67,93 +68,112 @@ static String symbolToString(const CSSValue* value)
+        return { };
+    CSSValueList* symbols;
+    if (!(symbols = downcast<CSSValueList>(symbolsValues.get())))
         return { };
```
consumeCounterStyleSymbols suggests this is always a list too. The same idiom applies here too.

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -61,99 +62,118 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
+            ASSERT_NOT_REACHED();
+            return { };
+        return result;
+        }
```
The code here looks wrong. From what I can understand, you can only have a list of size 2 or a single value.
```suggestion
    if (auto list = dynamicDowncast<CSSValueList>(negative.get())) {
        ASSERT(list->size() == 2);
        result.m_prefix = symbolToString(list->item(0));
        result.m_suffix = symbolToString(list->item(1));
        return result;
     }
```

vitorroriz replied:
ops, you are right, thanks

nt1m commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -61,99 +62,112 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
+            auto symbol = symbolToString(&pair->second());
+            result.constructAndAppend(symbol, weight);
+        } else
+            return { };
```
I think all the non pair cases are rejected by `consumeCounterStyleAdditiveSymbols`, so you could just `auto& pair = downcast<CSSValuePair>(additiveSymbol);` directly and replace `pair->` with `pair.`, and remove the `else` branch.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -61,99 +62,108 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
+    if (!value)
+        return { };
+
+    auto& additiveSymbols = downcast<CSSValueList>(*value);
```
Don’t need a local variable for this. Can just put it into the for statement.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -61,99 +62,108 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
-        ASSERT_NOT_REACHED();
-        return CSSCounterStyleDescriptors::NegativeSymbols();
-    }
+    if (auto list = dynamicDowncast<CSSValueList>(negative.get())) {
```
This can be `*negative` instead of `negative.get()`.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -61,99 +62,108 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
+    for (auto& symbol : symbols) {
+        auto string = symbolToString(&symbol);
+        if (!string.isNull())
+            result.append((string));
```
The `((string))` here isn’t needed. Just `(string)` should do.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -61,99 +62,108 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
+    if (!symbolsValues)
         return { };
 
+    auto& symbols = downcast<CSSValueList>(*symbolsValues);
```
Don’t need a local variable for this. Can just put it into the for statement.

vitorroriz committed:
Implement counter-style additive-symbols translation
https://bugs.webkit.org/show_bug.cgi?id=252926
rdar://103837257
Reviewed by Darin Adler and Tim Nguyen.

We are missing translation of the counter-styles
additive-symbols for the CSSCounterStylesDescriptor.

We are also introducing some fixes and style
adequation to the other translation functions.

* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
(WebCore::translateRangeFromStyleProperties):
(WebCore::symbolToString):
(WebCore::translateAdditiveSymbolsFromStyleProperties):
(WebCore::translatePadFromStyleProperties):
(WebCore::translateNegativeSymbolsFromStyleProperties):
(WebCore::translateSymbolsFromStyleProperties):
(WebCore::translateFallbackNameFromStyleProperties):
(WebCore::translatePrefixFromStyleProperties):
(WebCore::translateSuffixFromStyleProperties):
(WebCore::extractDataFromSystemDescriptor):
(WebCore::CSSCounterStyleDescriptors::create):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeCounterStyleAdditiveSymbols):

Canonical link: https://commits.webkit.org/260912@main
